### DISCUSSION TITLE
You are trying to `self.log()` but it is not managed by the `Trainer` control flow

### DISCUSSION THREAD

**Discussion by nian-liu** [Lightning-AI/pytorch-lightning#12660] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hello everyone,

I am new to pytorch-lighting and I have the following error with a simple classification code:
![image](https://user-images.githubusercontent.com/5463229/162238197-10437f33-3f87-4d12-af0f-dadc2c1a7bd1.png)
I am using self.log() like this:
![image](https://user-images.githubusercontent.com/5463229/162238493-9a7f0ed8-aff0-4ee3-85e2-327cd53aa8c4.png)

This error still occurs even I comment all these self.log() lines. Any suggestion will be appreciated. Thanks!

**Comment by rohitgr7** [comment]
hey @nian-liu !

are you sure these are all the `self.log` calls you have in your code? since it looks correct.

if possible can you share your code? or a reproducible script using https://colab.research.google.com/github/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report/bug_report_model.ipynb

**Reply by nian-liu** [reply]
Hi @rohitgr7,

I found where the problem is. 
Using the flag `log_gpu_memory='all'` in pl.Trainer leads to conducting `self._log_gpus_metrics()` in logger_connector.py, which may be "not managed by the `Trainer` control flow". After removing this flag, the error is gone. 
![image](https://user-images.githubusercontent.com/5463229/162311912-4531ebe1-0035-4e8e-b1ba-c6ea06727d68.png)

**Reply by rohitgr7** [reply]
this flag was removed recently. You can use `DeviceStatsMonitor` instead to log GPU memory stats.
